caching/base.py

```diff
@@ -57,11 +63,20 @@ def no_cache(self):
         return self.cache(config.NO_CACHE)
 
 
-class CacheInternalCommonMixin(object):
+class CachingModelIterable(ModelIterable):
     """
-    A set of methods common to our Django 1.8 and Django 1.9+ iterators.
+    Handles all the cache management for a QuerySet.
+
+    Takes the string representation of a query and a function that can be
```

Comment:
This part of the comment needs to be updated. Maybe: "Takes a queryset, and optionally takes a function that can be called to get an iterator over some database results. The function is only needed for RawQuerySets currently." (or something like that)